project/profiles/capacity4more/modules/c4m/og/c4m_og_invite_visitors/c4m_og_invite_visitors.module

```diff
@@ -0,0 +1,395 @@
+ * _c4m_og_invite_visitors_mail_text() to set up some additional tokens that can
+ * be used in email messages generated by c4m_og_invite_visitors_mail().
+ *
+ * @param array $replacements
+ *   An associative array variable containing mappings from token names to
+ *   values (for use with strtr()).
+ * @param array $data
+ *   An associative array of token replacement values. If the 'user' element
+ *   exists, it must contain a user account object with the following
+ *   properties:
+ *   - login: The UNIX timestamp of the user's last login.
+ *   - pass: The hashed account login password.
+ * @param array $options
+ *   Unused parameter required by the token_replace() function.
+ */
+function c4m_og_invite_visitors_mail_tokens(&$replacements, $data, $options) {
+  global $user, $base_url;
+  if (empty($data['user'])) {
+    return;
+  }
+  $group = c4m_og_current_group();
+  $url_options = array(
+    'absolute' => TRUE,
+    'purl' => array(
+      'provider' => 'og_purl|node',
+      'id' => $group->nid,
+    ),
+  );
+
+  $replacements['[user:full_name]'] = c4m_user_name($user);
+  $replacements['[user:public-profile-url]'] = $base_url . '/users/' . preg_replace('@[^a-z0-9-]+@', '-', strtolower(trim($replacements['[user:full_name]'])));
+  $replacements['[group:node_type]'] = $group->type;
+  $replacements['[group:node_title]'] = $group->title;
+  $replacements['[group:accept_invite_url]'] = user_pass_reset_url($data['user']);
+  $replacements['[group:dashboard_url]'] = url('<front>', $url_options);
+  $replacements['[personal:subject]'] = $data['form_fields']['subject'];
+  $replacements['[personal:body]'] = $data['form_fields']['body'];
+  $replacements['[site:name]'] = variable_get('site_name', 'Capacity 4 More');
+}
+
+/**
+ * Create new user and send email.
+ *
+ * @param string $invitee_email
+ *    Email address for the new user.
+ * @param array $form_fields
+ *    The invite form fields including a personal subject and message.
+ *
+ * @return object $account
+ *    A fully loaded user account.
+ */
+function _c4m_og_invite_visitors_create_user($invitee_email, $form_fields) {
+  // This will generate a random password.
+  $password = user_password(8);
+
+  // Username from email.
+  $name = explode('@', $invitee_email);
+
+  // To prepare the email address as a username, trim any potential leading
+  // and trailing spaces and replace simple illegal characters with hyphens.
+  // @see user_validate_name().
+  $name = preg_replace('/[^\x{80}-\x{F7} a-z0-9@_.\'-]/i', '-', trim($name[0]));
+
+  // Check if user name is available and append _NUM if existing.
+  if (user_load_by_name($name)) {
+    $i = 1;
+    $new_name = $name . '_' . $i;
+    while (user_load_by_name($new_name)) {
+      $i++;
+      $new_name = $name . '_' . $i;
+    }
+    $name = $new_name;
+  }
+
+  // Set up the user fields.
+  $fields = array(
+    'name' => $name,
+    'mail' => $invitee_email,
+    'pass' => $password,
+    'status' => 1,
```

Comment:
Yes but I thought since the invitation arrives by email, it serves the same purpose as an authentication email doesn't it? Sending another email is't necessary.